projects/ansible/build.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash -eu
+# Copyright 2021 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+
+# Change config = ConfigManager()
+# to
+# config = ConfigManager("/tmp/base.yml")
+# 
+# The init function of ConfigManager is executed when
+# a package that uses the ConfigManager is imported.
+# We therefore create /tmp/base.yml at fuzz-time.
+sed 's/config = ConfigManager()/config = ConfigManager(defs_file="\/tmp\/base.yml")/g' -i /src/ansible/lib/ansible/constants.py
+
+# Change 'with open(to_bytes(metadata_path))'
+# to
+# 'with open("/tmp/ansible_builtint_runtime.yml")'
+sed 's/to_bytes(metadata_path)/"\/tmp\/ansible_builtin_runtime.yml"/g' -i /src/ansible/lib/ansible/utils/collection_loader/_collection_finder.py
```

Comment:
What is the purpose of modifying these built-in static files?

Reply:
The reason here is that the two static files, `base.yml` and `ansible_builtin_runtime.yml`, won't exist when running the fuzzers through OSS-fuzz. So we specify the path of these two files before building Ansible, and each time we run a fuzzer we create the static file at that path. Thus, when we run our fuzzers, the two `.yml` files exist.

Reply:
Can you elaborate on why those files won't exist?

Reply:
Sure. This happens because of the way that OSS-fuzz uses Docker. When OSS-fuzz builds the fuzzers of a project the end result is an executable file. The Docker image used to build the fuzzers along with all of the projects source-files are deleted after the build is complete. The static files exist during the building of the fuzzers and the build will succeed, but when we run the fuzzer (which is the executable file we have created during the build-phase), the static files no longer exist.
Thus, when the executable attempts to open a static/configuration file like the `.yml` configuration files in the case of Ansible, we can expect an error, since the files don't exist.